### DISCUSSION TITLE
Persistent pinned tabs across windows/sessions

### DISCUSSION THREAD

**Discussion by imJoshuaRice** [zen-browser/desktop#1461] [Ideas]
I've just noticed I'm not alone in that pinned tabs can be entirely lost - and having looked intensively at both the Issues here and the Discord Server, I can see a few mentions of this being a Firefox specific issue.

There are a number of issues on this repository that mention losing pinned tabs, so I believe these could all be resolved via a more robust way of handling pinned tab persistence.

Two Discord Threads on the subject:

- [1280430564188360755](https://discord.com/channels/1088172780480114748/1280430564188360755) links to https://superuser.com/questions/1825979/firefox-loses-pinned-tabs 
- [1278291905553104926](https://discord.com/channels/1088172780480114748/1278291905553104926) mentions an almost identical setup in terms of extensions/addons and behavior.

I stumbled across the [PersistentPin](https://addons.mozilla.org/en-US/firefox/addon/persistentpin/) extension/addon a week or so ago and, since installing it, have not lost pinned tabs which I otherwise would lose after a day or two.

Whilst PersistentPin is working for me, I am wary that the it does not have an incredibly high user-count and I would feel much safer if it were a native feature of Zen.

I consider this Firefox behavior somewhat of a 'bug' that can be 'bugfixed' with an enhancement to Zen's functionality as well!

Thanks

**Comment by BoscoDomingo** [comment]
#2949

**Reply by imJoshuaRice** [reply]
> #2949

This is not the same issue..


This particular issue occurs when you open a new window - and close the new window last. When you re-open, only the last window is restored - causing you to lose all pinned tabs.
The solution to this issue is to either always make sure that your pinned tabs window is closed last, or use the restore window button.


That said, I moved back to Firefox after a plethora of other issues with Zen - notably the signing issue that has been ongoing for months, the inability to properly move pins which has been a problem since release and certain releases that have completely crippled the browser in the past (glance issue and then unloading tab issue).

**Reply by BoscoDomingo** [reply]
Oh I just wrote it here as a related issue to help organise similar matters with pinned tabs. In any case, sad to hear you had to leave Zen, but understandable since it's extremely unstable right now
